### PR TITLE
feat: Readonly view of course

### PR DESCRIPTION
**Pull request by piotr-pajak**
## Jira issue(s)
[LC-233](https://selleolabs.atlassian.net/browse/LC-233), [LC-234](https://selleolabs.atlassian.net/browse/LC-234), [LC-231](https://selleolabs.atlassian.net/browse/LC-231)

## Description
- Added the enrolled (boolean) field to getAllCourses
- Added the isAdmin parameter to the lesson service to allow the admin to view unassigned lessons
- Restricted views based on the user's role

[LC-233]: https://selleolabs.atlassian.net/browse/LC-233?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LC-234]: https://selleolabs.atlassian.net/browse/LC-234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LC-231]: https://selleolabs.atlassian.net/browse/LC-231?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/api/src/courses/api/courses.controller.ts [hunk 1]
```diff
@@ -49,6 +52,7 @@ export class CoursesController {
     @Query("page") page: number,
     @Query("perPage") perPage: number,
     @Query("sort") sort: SortCourseFieldsOptions,
+    @Query("userId") userId: string,
```
Comment (wielopolski) [inline apps/api/src/courses/api/courses.controller.ts:55]:
```suggestion
    @CurrentUser("userId") currentUserId: string,
```
Api cannot rely on information from FE regarding userId.

File: apps/api/src/lessons/api/lessons.controller.ts [hunk 2]
```diff
@@ -14,15 +17,20 @@ export class LessonsController {
...
+      {
+        type: "query",
+        name: "id",
+        schema: Type.String({ format: "uuid" }),
+      },
     ],
   })
   async getLesson(
     @Query("id") id: string,
+    @Query("isAdmin") isAdmin: boolean,
```
Comment (wielopolski) [inline apps/api/src/lessons/api/lessons.controller.ts:29]: Same here. 
```suggestion
    @CurrentUser("role") userRole: UserRole,
```
or
```suggestion
       @CurrentUser() currentUser: { userId: string, role: string },
```

**Review by wielopolski — COMMENTED** [review]
GJ 🍻  Some suggestions 😄

**Review by typeWolffo — APPROVED** [review]
:shipit:

**Review by wielopolski — APPROVED** [review]
GJ 🍻

**Pull request merged**